src/test/resources/a/com/example/ImportTest.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2018 Cadix Development (https://www.cadixdev.org)
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which accompanies this distribution,
+ * and is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package com.example;
+
+import com.example.other.AnotherClass;
+import static com.example.pkg.Constants.*;
+import com.example.other.OtherClass;
+import java.lang.String;
+import java.lang.Exception;
```

Comment:
I think the problem is here: the static import is mixed with the other imports. Mercury handles normal imports through the `ImportRewrite`, but static imports through the normal remapping. That's why the `TextEdit`s overlap.

I suppose your workaround is fine for now if it avoids the problem, but in general we would also need to handle static imports through the `ImportRewrite`. Theoretically it is possible that you remap members in such a way that a remapped static import conflicts with some other remapped members in a class. Such conflicts are only handled within `ImportRewrite`.

Reply:
Do you think this is a reasonable "for-now" change to have merged?

Reply:
It's probably fine, maybe with an extra comment that shows that this workaround should be removed again once someone manages to fix up all the import mess.